lua/lazyvim/plugins/extras/lang/nix.lua

```diff
@@ -0,0 +1,35 @@
+return {
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      opts.ensure_installed = opts.ensure_installed or {}
+      vim.list_extend(opts.ensure_installed, { "nix" })
+    end,
+  },
+  {
+    "williamboman/mason.nvim",
+    opts = function(_, opts)
+      vim.list_extend(opts.ensure_installed, {
+        "nixpkgs-fmt",
+        "nil",
```

Comment:
Both of these concerns (formatter, LSP) have at least two successful, widespread community options and are highly subjective choices, particularly the formatter. I know a lot of folks really like `alejandra` for example and it produces _very_ different outcomes than `nixpkgs-fmt`. For flakes-based projects, IMO it's better to set `outputs.formatter.${system}` to the preferred package and not have an $EDITOR-wide selection at all.

There's also an open RFC attempting to implement a one-true-formatting-standard outcome:

https://github.com/NixOS/rfcs/pull/166

All that to say I don't think having this extra choose one particular option for these concerns is going to satisfy a great cross-section of Nix users.

Reply:
I think having one formatter installed other than having no formatter installed is a better UX, we can remove it *when*/*if* there is a standard in the fragmented ecosystem. Users & non-NixOS home-manager users can always customize and remove these configurations. Please let me know if you'd like this issue to be reopened, I'm closing it now.